core/controllers/reader.py

```diff
@@ -79,26 +79,62 @@ def _does_exploration_exist(exploration_id, version, collection_id):
 class ExplorationEmbedPage(base.BaseHandler):
     """Page describing a single embedded exploration."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': editor.SCHEMA_FOR_EXPLORATION_ID
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'v': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 1
+                    }]
+                },
+                'default_value': None
+            },
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.ENTITY_ID_REGEX
+                    }]
+                },
+                'default_value': None
+            },
+            'iframed': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': False
+            },
+        }
+    }
+
     @acl_decorators.can_play_exploration
     def get(self, exploration_id):
         """Handles GET requests.
 
         Args:
             exploration_id: str. The ID of the exploration.
         """
-        version_str = self.request.get('v')
+        version_str = self.normalized_request.get('v')
```

Comment:
Why is this string? In the schema above you say it is int.

Reply:
I do not know why it is string.. I took 'v' as int because of this
https://github.com/oppia/oppia/blob/24bb4674c5430954aa9c50bc79f562dc9a7facc9/core/controllers/reader.py#L58

Reply:
I think it might be string because v can be None and if v is None, it would be empty and that would be a string

Reply:
Can you run the code and verify what is the type of this?

Reply:
It shows that `version_str` is `string` and `version` is `int`

Reply:
Here version is part of the url query parameters, and it is expected to be in string as [requested from here](https://github.com/oppia/oppia/blob/develop/core/controllers/reader.py#L176). 
But this line should be removed becuase by normalization string is automatically converted into int if its schema is expecting a int type. [reference](https://github.com/oppia/oppia/blob/00282e533b5832cb763100de1a5cc727644d64ef/core/schema_utils.py#L147)

Reply:
Instead of line 125 and 126 this should be added --
`version = self.normalized_request.get('v')`